### PR TITLE
#169258566 Configure react application

### PR DESCRIPTION
**Pull request by NiyongaboEric**
#### What does this PR do?
The PR help us to set up react application from scratch.

#### Description of Task to be completed?
  - added folder structure
  - added NPM package
  - configured Webpack
  - added Babel
  - added Eslint
  - add Pull request template
  - [Finish #169258566]

#### How should this be manually tested?
In order to test the application try the following steps
- `git clone ch-setup-application-169258566`
- `cd team-odd-bn-frontend`
- `git checkout ch-setup-application-169258566`
- `npm install`
- `npm run dev` 

#### What are the relevant pivotal tracker stories?
[#169258566](https://www.pivotaltracker.com/story/show/169258566)
#### Screenshots (if appropriate)
Development
Screen Shot 2019-12-23 at 13 13 41

Production
Screen Shot 2019-12-24 at 14 25 25

#### Questions:

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        },
+        "ecmaVersion": 2018,
+        "sourceType": "module"
+    },
+    "plugins": [
+        "react"
+    ],
+    "rules": {
+    }
+}
```
Comment (nziokaivy) [inline .eslintrc.json]: EOL

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "trailingComma": "es5",
+  "tabWidth": 2,
+  "semi": false,
+  "singleQuote": true
+}
```
Comment (nziokaivy) [inline .prettierrc]: End of line

**Review by nziokaivy — CHANGES_REQUESTED** [review]
Good Job @NiyongaboEric 
However, please take note of the end of line in every file.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "dev": "webpack-dev-server --mode development --hot --open --config webpack.config.js",
+    "build": "webpack --mode production",
+    "start": "HTTPS=true serve -s ./dist"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/andela/team-odd-bn-frontend"
+  },
+  "author": "Eric Niyongabo",
```
Comment (hezronkimutai) [inline package.json]: @NiyongaboEric  Change this to team Odd

File: package.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/andela/team-odd-bn-frontend/issues"
+  },
+  "homepage": "https://github.com/andela/team-odd-bn-frontend#readme",
+  "dependencies": {
+    "@babel/core": "^7.7.5",
+    "@babel/preset-env": "^7.7.6",
+    "@babel/preset-react": "^7.7.4",
```
Comment (hezronkimutai) [inline package.json]: I have a question concerning babel, is there need to have it in the dependencies yet the application we run in heroku is already build, I mean is already converted to es5

File: public/index.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+  <html lang="en">
+  <head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Crime-App</title>
```
Comment (hezronkimutai) [inline public/index.html]: @NiyongaboEric  Change the title of this App too to Barefoot nomad.

**Review by hezronkimutai — CHANGES_REQUESTED** [review]
Nice work Eric, I have left some comments please feel free to view it an make the changes

File: package.json (hunk 4 above)
Comment (NiyongaboEric) [reply package.json]: You are right @hezronkimutai.

**Review by hezronkimutai — APPROVED** [review]
Nice work @NiyongaboEric

**Review by nziokaivy — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]
Good job @NiyoEric 
You've built a good foundation we'll all be using.

**Review by william000000 — APPROVED** [review]
Nice Work @NiyoEric, I have noticed that you also set up eslint and prettier, That is great!

**Review by shemaeric — CHANGES_REQUESTED** [review]
This is a good work @NiyongaboEric . Though I would like to suggest looking at how you can change the codes in `webpack.config.js` into ES6 codes.

**Review by shemaeric — APPROVED** [review]

**Pull request merged**